crates/bioauth-flow-rpc/src/lib.rs

```diff
@@ -366,14 +369,56 @@ where
             )
             .map_err(AuthenticateError::RuntimeApi).map_err(errtype)?;
 
-        self.pool
+        info!("Bioauth flow - authenticate transaction is sending");
+
+        let mut watch = self.pool
             .submit_and_watch(
                 &sp_api::BlockId::Hash(at),
                 sp_runtime::transaction_validity::TransactionSource::Local,
                 ext,
             )
             .await
-            .map_err(AuthenticateError::BioauthTx).map_err(errtype)?;
+            .map_err(AuthenticateError::BioauthTx).map_err(errtype)?.fuse();
+
+        loop {
+            let tx_status = watch.select_next_some().await;
```

Comment:
I don't get it, why not just use `next()`?

Reply:
`select_next_some` will do the same as `next().await.unwrap()` - panic if empty stream is pulled

Reply:
Agree - https://github.com/humanode-network/humanode/pull/1018/commits/348e3683516bccbd809a5a52a99d77023b053113.